spring-integration-core/src/main/java/org/springframework/integration/support/management/ExponentialMovingAverage.java

```diff
@@ -69,18 +87,52 @@ public synchronized void reset() {
 	 * @param value the measurement to append
 	 */
 	public synchronized void append(double value) {
-		if (value > max || count == 0) {
-			max = value;
-		}
-		if (value < min || count == 0) {
-			min = value;
+		if (this.samples.size() == this.retention) {
+			samples.remove(0);
 		}
-		sum = decay * sum + value;
-		sumSquares = decay * sumSquares + value * value;
-		weight = decay * weight + 1;
+		samples.add(value);
 		count++;//NOSONAR - false positive, we're synchronized
 	}
 
+	public Statistics calc() {
+		List<Double> copy;
+		long count;
+		synchronized (this) {
```

Comment:
The idea is to have minimal impact on `append()` (writers) which has to be synchronized too.